web-app/src/components/BookingModal/index.js

```diff
@@ -54,31 +86,33 @@ const BookingModal = props => {
             toggleRejectionMessageInputView={toggleRejectionMessageInputView}
           />
         )}
-        <h1>
-          {isEventBeingUpdated ? 'Update Booking' : 'Request a Holiday'}
+        {toggleRejectionMessageView && (<div><h1>Message History</h1>{legacyHolidayMessagelist(legacyMessages)}</div>)}
+        {!toggleRejectionMessageView && (<div><h1>
```

Comment:
Pull what you're rendering in these out into their own functions so there's no so many elements relying on an in-line conditional check